CONTRIBUTING.md

```diff
@@ -1 +1,172 @@
+  class NeatFeatureType(Enum):
+    '''
+    Enum containing available features of this type.
+    '''
+    Feature1 = auto()
+    Feature2 = auto()
+    NeatFeature = auto()
+   ```
+Lastly, the user needs to use the setter function for the feature type to take the appropriate parameter from the input deck and instantiate the corresponding class. These functions are located in the `src/<package-name>/tools.py` file and always start with `set_<name_of_instantiated_object>`. For example, we could have a function called `set_neatfeatures` which would look like this:
+
+  ```python
+  import numerics.neatfeatures as neat_feature_defs
+  from general import NeatFeatureType
+  
+  def set_neatfeatures(params):
+	'''
+	Given the NeatFeature parameter, set the neat_feature object
+
+	Inputs:
+	-------
+		params: list of parameters from solver
+
+	Outputs:
+	--------
+	    neat_feature: instantiated neat_feature object
+	'''
+	feature_name = params["NeatFeature"]
+	if NeatFeatureType[feature_name] == NeatFeatureType.NeatFeature:
+		neat_feature = neat_feature_defs.NeatFeature()
+	else:
+		raise NotImplementedError("Feature not supported")
+	return neat_feature
+  ```
+  
+These interfaces allow the user to focus on the meat of their new feature while not having too much overhead in making the feature available to the users.
+
+## Style Guide
+
+#### Some general guidelines to follow are:
+* Use tabs with a length of four spaces
+* Indent continued linestwice
+* Limit lines to a maximum of 78 characters
+For more detailed guidelines, please refer to the [PEP 8 style guide for Python code](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds), which we largely aim to follow.
+#### Commenting Style
+* Class commenting style
+  ```python
+  class ClassName():
+  	'''
+  	Brief description of the class
+  
+  	Attributes:
+  	-----------
+  	attribute name: brief description
+  
+  	Abstract Methods:
+  	-----------------
+  	abstract_method_name
+  		description of the abstract method
+  	'''
+  ```
+* Function definitions
+  ```python
+  def function_name(**args):
+  	'''
+  	Brief description of function
+  
+  	Inputs:
+  	-------
+  	input name: brief description [array shape if available]
+
+  	Outputs:
+  	--------
+  	output name: brief description [array shape if available]
+
+  	Notes:
+  	------
+  	additional notes and comments can be added here if needed
+  	'''
+  ```
+## Branch naming conventions
+
+Quail uses the following branch naming conventions:
+`<branch_type_name>/<branch_name>`
+There are multiple branch types that are standard in Quail. These include the following:
+* `bug`: these branches are designed for identifying, isolating, and fixing bugs that may exist throughout Quail. 
+* `feature`: these branches are designed for the addition of new features.
+* `test`: branches used for the specific addition of unit tests (tests can also be added throughout feature dev in the appropriate branch).
```

Comment:
the wording inside the parentheses is kind of confusing

Reply:
agreed. I have removed this. It is trying to say that a test can be added in any kind of branch and that we don't have to make specific test only branches, but I think this is not so necessary.